hatch_jupyter_builder/utils.py

```diff
@@ -94,25 +94,27 @@ def npm_builder(
     if isinstance(npm, str):
         npm = [npm]
 
+    is_yarn = (abs_path / "yarn.lock").exists()
+
     # Find a suitable default for the npm command.
     if npm is None:
-        is_yarn = (abs_path / "yarn.lock").exists()
         if is_yarn and not which("yarn"):
             log.warning("yarn not found, ignoring yarn.lock file")
             is_yarn = False
-
         npm = ["yarn"] if is_yarn else ["npm"]
 
     npm_cmd = normalize_cmd(npm)
 
+    install_cmd = ["install", "--immutable"] if is_yarn else ["install", "--no-audit", "--no-fund"]
```

Comment:
It might be reasonable to use [`npm ci`](https://docs.npmjs.com/cli/v7/commands/npm-ci) instead of `npm install`. `npm ci` is intended for automated environments where determinism is crucial, and it's roughly equivalent to `yarn install --immutable`.

However, `npm ci` does remove and reinstall `node_modules`. With package download caches, this doesn't use additional bandwidth, but it's definitely slower than `npm install`.

Do we care more about correctness or speed when building internal JS assets for a Python package?

Reply:
I'm wary of changing any of this default behavior, as I'm not a JS developer anymore and I'm not sure what affect this would have on existing consumers.

Reply:
Ok, are you comfortable with this change as-is?
* `--no-fund` only changes what's written to stdout, to make it cleaner in logs
* `--no-audit` speeds up the command by preventing a background HTTP request and prevents additional output to stdout
* For Yarn, [`--immetable` is already the behavior when this is run in CI](https://yarnpkg.com/cli/install) (determined via environment variables), so explicitly setting it makes the hook more consistent across all environments. Substantively, it's an important check, as it ensures that the `yarn.lock` will not be modified by the hook, which is very important to deterministic package builds.